File: Native/Test/Python/mr-reduce.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric

client = fabric.createClient()

cv = client.MR.createConstValue("Size", 10000000)

ago = client.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Float64 output, Size index) { output = Float64(index+1); }", "foo")
ag = client.MR.createArrayGenerator(cv, ago)

mo = client.KLC.createArrayMapOperator("bar.kl", "operator bar(Float64 input, io Float64 output) { output = 1.0 / (input * input); }", "bar")
m = client.MR.createArrayMap(ag, mo)

ro = client.KLC.createReduceOperator("reduce.kl", "operator reduce(Float64 input, io Float64 result) { result += input; }", "reduce")
r = client.MR.createReduce(m, ro)

print(r.produce())

client.close()

File: Native/Test/Node/mr-reduce.js
/*
 *  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
 */

FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10000000);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Float64 output, Size index) { output = Float64(index+1); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

mo = FC.KLC.createArrayMapOperator("bar.kl", "operator bar(Float64 input, io Float64 output) { output = 1.0 / (input * input); }", "bar");
m = FC.MR.createArrayMap(ag, mo);

ro = FC.KLC.createReduceOperator("reduce.kl", "operator reduce(Float64 input, io Float64 result) { result += input; }", "reduce");
r = FC.MR.createReduce(m, ro);

console.log(r.produce());

FC.close();
